### DISCUSSION TITLE
Clarification about sign of coefficients in spinless Fermion basis

### DISCUSSION THREAD

**Discussion by archisman-panigrahi** [QuSpin/QuSpin#512] [Q&A] [answered]
I am following the example in the [documentation](https://weinbe58.github.io/QuSpin/generated/quspin.basis.spinless_fermion_basis_1d.html#quspin.basis.spinless_fermion_basis_1d)

```
from quspin.operators import hamiltonian # Hamiltonians and operators
from quspin.basis import spinless_fermion_basis_1d # Hilbert space spinless fermion basis
import numpy as np # generic math functions
#
##### define model parameters #####
L=10 # system size
J=1.0 # hopping strength
mu=0.9045 # chemical potential
U=1.5 # nn interaction strength
##### define periodic drive #####
Omega=4.5 # drive frequency
def drive(t,Omega):
	return np.cos(Omega*t)
drive_args=[Omega]
#
##### construct basis in the 0-total momentum and +1-parity sector
basis=spinless_fermion_basis_1d(L=L,Nf=L//2,a=1,kblock=0,pblock=1)
print(basis)
# define PBC site-coupling lists for operators
n_pot=[[-mu,i] for i in range(L)]
J_nn_right=[[-J,i,(i+1)%L] for i in range(L)] # PBC
J_nn_left=[[+J,i,(i+1)%L] for i in range(L)] # PBC
U_nn_int=[[U,i,(i+1)%L] for i in range(L)] # PBC
# static and dynamic lists
static=[["+-",J_nn_left],["-+",J_nn_right],["n",n_pot]]
dynamic=[["nn",U_nn_int,drive,drive_args]]
###### construct Hamiltonian
H=hamiltonian(static,dynamic,dtype=np.float64,basis=basis)
``` 

The Hamiltonian is, 
![image](https://user-images.githubusercontent.com/3063132/158443926-7c6fd9f4-e997-40e6-8194-e8f8b30ec93a.png)

The Hamiltonian has -J in the coefficients of both the terms c_{j+1} c^{\dagger}_j and also c_{j} c^{\dagger}_{j+1}.

It is unclear to me why two different signs, -J and +J are used in the two lines

J_nn_right=[[**-J**,i,(i+1)%L] for i in range(L)] # PBC
J_nn_left=[[**+J**,i,(i+1)%L] for i in range(L)] # PBC

If I use -J in both these lines, then I get a warning that the Hamiltonian is non-Hermitian.

**Comment by weinbe58** [comment, marked as answer]
The hermtian conjugate changes the order of the operators in the hopping
terms but in the hopping list the order is reversed, hence the extra
negative sign.
